.travis.yml

```diff
@@ -45,10 +45,8 @@ android:
   - adb shell input keyevent 82 #Unlock phone before using
 jobs:
   include:
-  - stage: Build
-    script: ./gradlew clean build -xlint -xtest -PdisablePreDex $PROPERTIES
-  - stage: checkstyle
-    script: ./gradlew checkstyle -xlint -xtest  $PROPERTIES
+  - stage: Validation
+    script: ./gradlew checkstyle pmd clean findbugs -xlint -xtest $PROPERTIES
```

Comment:
I'm not sure you need the `clean` here. Likewise for `-xlint` and `-xtest`. I suggest trying a `./gradlew <task> --dry-run` for each of the validation tasks you'd like to be executed. If you find that lint and test are on the task graph, there may be an issue with the task dependencies; I would not expect checkstyle, pmd, or findbugs to depend on lint or test.

Reply:
When testing locally, I got the problem with `Findbugs` task. It always inspects the old code. `clean` solves my problem. Also, `Findbugs` depends on `Assemble`.
Do you think we need to separate `findbugs` on a different stage? and remove `-xlint` and `-xtest` in `Validation` stage